crates/flox-rust-sdk/src/models/environment/mod.rs

```diff
@@ -304,18 +306,21 @@ pub struct PkgDbError {
     /// The generic message for this category of error.
     pub category_message: String,
     /// The more contextual message for the specific error that occurred.
-    pub context_message: String,
+    pub context_message: Option<String>,
     /// The underlying error message if an exception was caught.
-    pub caught_message: String,
+    pub caught_message: Option<String>,
 }
 
 impl Display for PkgDbError {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(
-            f,
-            "{}: {}: {}",
-            self.category_message, self.context_message, self.caught_message
-        )
+        write!(f, "{}", self.category_message)?;
+        if let Some(context_message) = &self.context_message {
+            write!(f, ": {}", context_message)?;
+        }
+        if let Some(caught_message) = &self.caught_message {
+            write!(f, ": {}", caught_message)?;
+        }
```

Comment:
suggestion non-blocking: a bit more "rustic" version of this

```suggestion
        if let Some(ref context_message) = self.context_message {
            write!(f, ": {}", context_message)?;
        }
        if let Some(ref caught_message) = self.caught_message {
            write!(f, ": {}", caught_message)?;
        }
```